pkg/controller/ingress/concurrency_watchdog.go

```diff
@@ -336,3 +346,23 @@ func podIsReady(pod *corev1.Pod) bool {
 	}
 	return false
 }
+
+// podIsActive checks if a Pod is Ready and exists and is able to serve connections
+func podIsActive(ctx context.Context, lgr logr.Logger, cl client.Client, pod client.ObjectKey) (bool, error) {
+	var obj corev1.Pod
+	err := cl.Get(ctx, pod, &obj)
+	if apierrors.IsNotFound(err) {
+		lgr.Info("pod doesn't exist anymore")
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("getting pod: %w", err)
```

Comment:
could also log error here so as to keep everything related to active check completely localized

Reply:
Pushed an update to just log here. I like that